sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -5292,6 +5292,183 @@ object functions {
    */
   def hours(e: Column): Column = withExpr { Hours(e.expr) }
 
+  /**
+   * Creates a map after splitting the text into key/value pairs using delimiters.
+   * Both `pairDelim` and `keyValueDelim` are treated as regular expressions.
+   *
+   * @group map_funcs
+   * @since 3.5.0
+   */
+  def str_to_map(text: Column, pairDelim: Column, keyValueDelim: Column): Column = withExpr {
+    StringToMap(text.expr, pairDelim.expr, keyValueDelim.expr)
+  }
+
+  /**
+   * Creates a map after splitting the text into key/value pairs using delimiters.
+   * Default delimiter is ':' for `keyValueDelim`.
+   * The `pairDelim` is treated as regular expressions.
+   *
+   * @group map_funcs
+   * @since 3.5.0
+   */
+  def str_to_map(text: Column, pairDelim: Column): Column = withExpr {
```

Comment:
`def str_to_map(text: Column, pairDelim: Column, keyValueDelim: Column): Column` is missing?